docs/database/database-spans.md

```diff
@@ -85,6 +87,56 @@ and SHOULD adhere to one of the following values, provided they are accessible:
 If a corresponding `{target}` value is not available for a specific operation, the instrumentation SHOULD omit the `{target}`.
 For example, for an operation describing SQL query on an anonymous table like `SELECT * FROM (SELECT * FROM table) t`, span name should be `SELECT`.
 
+## Status
+
+[Span Status Code][SpanStatus] MUST be left unset if the operation has ended without any errors.
+
+Instrumentation SHOULD consider the operation as failed if any of the following is true:
+
+- the `db.response.status_code` value indicates an error
+
+  > [!NOTE]
+  > The classification of status code as an error depends on the context.
+  > For example, a SQL STATE `02000` (`no_data`) indicates an error when the application
+  > expected the data to be available. However, it is not an error when the
+  > application is simply checking whether the data exists.
+  >
+  > Instrumentations that have additional context about a specific operation MAY use
+  > this context to set the span status more precisely.
+  > Instrumentations that don't have any additional context MUST follow the
+  > guidelines in this section.
+
+- an exception is thrown by the instrumented method call
+- the instrumented method returns an error in another way
+
+When the operation ends with an error, instrumentation:
+
+- SHOULD set the span status code to `Error`
+- SHOULD set the `error.type` attribute
+- SHOULD set the span status description when it has additional information
```

Comment:
should this have any requirements about not having any confidential data? Similar to how we have on metrics errors where we don't add if it shows actual values of queries, etc

Reply:
I don't see how database instrumentation can determine if there is any sensitive or confidential data in the arbitrary exception message.  Someone can put sensitive data into database name (bluesky has DB per user and they might call them accordingly), but how can we provide basic observability in the common case if we assume the worst? 

Happy to consider any specific cases when we can do something on the instrumentation side.

Reply:
Sorry if I wasn't clear, I didn't mean that is the job of the instrumentation to determine this, I meant more as a recommendation, that this sentence could include something on the lines of 

```
SHOULD set the span status description when it has additional information about the error that aligns with [Span Status Description][SpanStatus] and does not contain sensitive data.
```